cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -122,6 +133,24 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     String subnet = conf.getSubnet();
     Network networkInfo = getNetworkInfo(projectId, network, compute);
 
+    boolean useInternalIP;
+    
+    // if a peered network is specified then the instance is tied to a network
+    PeeringState state = peeringState(systemNetwork, systemProjectId, networkInfo);
+    if (state == PeeringState.ACTIVE) {
+      if (conf.isPreferExternalIP()) {
+        LOG.warn("Prefer external IP setting will be ignored and internal IP will used to communicate with the " +
+                   "Dataproc cluster as active peering with the system network found.");
+      }
+      useInternalIP = true;
+    } else if (state == PeeringState.INACTIVE) {
+      throw new IllegalStateException(String.format("VPC Peering to the system network %s in system project %s found " +
+                                                      "in INACTIVE state. Please fix peering setup to be in ACTIVE " +
+                                                      "state.", systemNetwork, systemProjectId));
+    } else {
+      useInternalIP = !conf.isPreferExternalIP();
```

Comment:
this is not correct. This setting is used to tell the provisioner to use the external IP even when it detects that the internal IP can be used. When this is false and the provisioner detects that the internal IP *cant* be used (when the system network does not exist or is different than the dataproc network and it is not peered), useInternalIP should be false.

Reply:
done. so conf.isPreferExternalIP takes precedence.